### PR TITLE
Sync head; fix default extensions; rename homeUnitId

### PR DESCRIPTION
**Pull request by shayne-fletcher**
- Sync https://gitlab.haskell.org/ghc/ghc.git `8a51b2ab7433c06bddca9699b0dfd8ab1d11879b`;
- Stop assuming the `default-extensions` field is static and populate it properly;
- `homeUnitId` -> `homeUnitId_`.

### CODE REVIEW

File: ghc-lib-gen/src/Ghclibgen.hs [hunk 1]
```diff
@@ -744,9 +741,8 @@ generateGhcLibParserCabal ghcFlavor = do
         , "    build-depends:"
         ] ++ indent2 (withCommas (commonBuildDepends ghcFlavor)) ++
         [ "    build-tools: alex >= 3.1, happy >= 1.19.4"
-        , "    other-extensions:"
-        ] ++
-        indent2 (askField lib "other-extensions:") ++
+        , "    other-extensions:" ] ++ indent2 (askField lib "other-extensions:") ++
+        [ "    default-extensions:" ] ++ indent2 (askField lib "default-extensions:") ++
```
Comment (hurryabit) [inline ghc-lib-gen/src/Ghclibgen.hs:745]: This looks like we have the same code twice. Is there a specific reason for this?

Comment (shayne-fletcher) [reply ghc-lib-gen/src/Ghclibgen.hs:745]: Two functions for two cabal files : `ghc-lib-parser.cabal` and `ghc-lib.cabal`. No deeper reason than that.

Comment (shayne-fletcher) [reply ghc-lib-gen/src/Ghclibgen.hs:745]: Unless I misunderstand your question and you are confusing lines 744 and 745 as being the same? (They're not they differ in field-name).

Comment (hurryabit) [reply ghc-lib-gen/src/Ghclibgen.hs:745]: Your first answer answers my question. Thanks a lot.

**Pull request merged**